packages/icons/src/utils.ts

```diff
@@ -1,8 +1,33 @@
-import { CryptoIconName, cryptoIcons, FlagIconName, flagIcons } from './icons';
+import { EthereumTokenSymbol } from '@suite-common/wallet-types';
 
-export const isCryptoIconType = (
-    iconName: CryptoIconName | FlagIconName,
-): iconName is CryptoIconName => iconName in cryptoIcons;
+import {
+    CryptoIconName,
+    cryptoIcons,
+    EthereumTokenIconName,
+    ethereumTokenIcons,
+    FlagIconName,
+    flagIcons,
+    IconName,
+    icons,
+} from './icons';
+import { IconNames } from './types';
 
-export const isFlagIconType = (iconName: CryptoIconName | FlagIconName): iconName is FlagIconName =>
+export const isCryptoIconType = (iconName: IconNames): iconName is CryptoIconName =>
+    iconName in cryptoIcons;
+
+export const isIconType = (iconName: IconNames): iconName is IconName => iconName in icons;
+
+export const isFlagIconType = (iconName: IconNames): iconName is FlagIconName =>
     iconName in flagIcons;
+
+// First we check whether the name is in the list of Ethereum token symbols.
+// Since we cannot guarantee this, if it's not, we default to the 'erc20' icon.
+// Then we check whether the name is in the list of Ethereum token icons.
+const getIsEthereumTokenIconUploaded = (name: EthereumTokenSymbol) =>
+    (name in ethereumTokenIcons ? name : 'erc20') as EthereumTokenIconName;
+export const isEthereumTokenIconType = (iconName: IconNames): iconName is EthereumTokenIconName => {
+    if (!getIsEthereumTokenIconUploaded(iconName as EthereumTokenSymbol)) {
```

Comment:
`getIsEthereumTokenIconUploaded(iconName as EthereumTokenSymbol)` will always return a string, thus this condition will never resolve to true. And even if it would execute the line `return 'erc20' in ethereumTokenIcons;` would be always `true`. Is this condition necessary? 

IMHO you could change it to something like this:

```ts
const isEthereumIconType = (iconName: IconNames) => iconName in ethereumTokenIcons;

const getEthereumTokenIcon = (name: EthereumTokenSymbol) =>
    (isEthereumIconType(name) ? name : 'erc20') as EthereumTokenIconName;
```